enumerables.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+# rubocop: disable Metrics/CyclomaticComplexity
+
+# rubocop: disable Metrics/PerceivedComplexity
+
+# rubocop: disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?(defualt = nil)
+    selfitem = self
+    if !block_given? && defualt.nil?
+      selfitem.my_each { |x| return false unless x }
+    elsif defualt.is_a? Regexp
+      selfitem.my_each { |x| return false unless x =~ defualt }
+    elsif block_given?
+      selfitem.my_each { |x| return false unless yield(x) }
+    else
+      selfitem.my_each { |x| return false unless x }
+    end
+    true
+  end
```

Comment:
`[1, 0, 2.1]..my_all?(Integer)` and `[1, 0, 2.1]..my_all?(3)` both return `true` instead of `false` You also check for exact values as well as classes.